glommio/src/sys/uring.rs

```diff
@@ -676,6 +676,7 @@ trait UringCommon {
     /// Return `None` if no event is completed, `Some(true)` for a task is woken
     /// up and `Some(false)` for not.
     fn consume_one_event(&mut self) -> Option<bool>;
+    fn wait_for_events(&mut self, events: u32) -> io::Result<()>;
```

Comment:
if we had a bug about cancellations not being sent, it would be good if this came with an accompanying test.

Reply:
I've tried to write a test that reliably (and cheaply) reproduces the bug for a few hours, but I've not been successful. 

The kernel may apply back-pressure and refuse our SQE submissions if it fails to allocate or if the CQE ring is full. If that happens before we submit the preemption timer, we panic because of a failed assertion (before this patch). 

Because I don't have control over the kernel, it's difficult for me to recreate those conditions.